### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by csgoh** [csgoh/roadmapper#11] [Q&A] [answered]
First major release

<hr /><em>This discussion was created from the release <a href='https://github.com/csgoh/roadmapper/releases/tag/1.0.0'>v1.0.0</a>.</em>

**Comment by bass4aj** [comment, marked as answer]
I was playing with the demo code and found the "with" statements in the demo01 method to be failing. looking through the code, it seems like the roadmap class is missing a context manager to allow the "with" statement to function. Is this correct or am I missing something?

**Reply by csgoh** [reply]
Hi, thanks for trying out roadmapper. I was experimenting 'with' statement earlier on but decided to drop it. It was somehow still in the demo code which I forgot to remove. If you find the 'with' statement useful, let me know, I can bring it back in the next release.

**Reply by bass4aj** [reply]
Ah that makes more sense. I do appreciate the ability to write cleaner code with it, but its not critical. I also found that roadmap.set_colour_palette in demo03 no longer works. had to rewrite to roadmap._Roadmap__set_colour_palette

**Reply by bass4aj** [reply]
I'm currently playing with the idea of using roadmapper in a python wizard that would import an excel template of data and convert it to a standard formatting for work so all the non-technical folks can stop using excel to build gantt chart visuals

**Reply by csgoh** [reply]
Do not refer to the code in the demo file. I was using it to test roadmapper and it contains older code. 
Please refer to the wiki: https://github.com/csgoh/roadmapper/wiki/Usage-Documentation

**Reply by csgoh** [reply]
Context manager 'with' syntax is now added to v1.2.0.
